tests/stockfish/test_models.py

```diff
@@ -448,9 +478,48 @@ def test_get_evaluation_stalemate(self, stockfish):
 
     def test_set_depth(self, stockfish):
         stockfish.set_depth(12)
-        assert stockfish.depth == "12"
+        assert stockfish._depth == 12
         stockfish.get_best_move()
         assert "depth 12" in stockfish.info
+        stockfish.set_depth()
+        assert stockfish._depth == 15
+        stockfish.get_best_move()
+        assert "depth 15" in stockfish.info
+
+    @pytest.mark.parametrize("depth", ["12", True, 12.1, 0, None])
+    def test_set_depth_raises_type_error(self, stockfish, depth):
+        with pytest.raises(TypeError):
+            stockfish.set_depth(depth)
+
+    def test_get_depth(self, stockfish):
+        stockfish.set_depth(12)
+        assert stockfish.get_depth() == 12
+        assert stockfish._depth == 12
+        stockfish.set_depth(20)
+        assert stockfish.get_depth() == 20
+        assert stockfish._depth == 20
+
+    def test_set_num_nodes(self, stockfish):
+        stockfish.set_num_nodes(100)
+        assert stockfish._num_nodes == 100
+        stockfish.set_num_nodes()
+        assert stockfish._num_nodes == 1000000
+
+    def test_set_num_nodes_raises_type_error(self, stockfish):
```

Comment:
Here I would also prefer to use `@pytest.mark.parametrize`. I must have overlooked this test last time.